featuretools/primitives/standard/datetime_transform_primitives.py

```diff
@@ -461,6 +461,58 @@ def is_weekend(vals):
         return is_weekend
 
 
+class IsYearEnd(TransformPrimitive):
+    """Determines if a date falls on the end of a year.
+
+    Examples:
+        >>> from datetime import datetime
+        >>> dates = [datetime(2019, 12, 31),
+        ...          datetime(2019, 1, 1),
+        ...          datetime(2019, 11, 30)]
```

Comment:
Can we include an example with NaN?

Reply:
We should make sure we did this with the previous Datetime primitives we merged in

Reply:
General question regarding the behavior of primitives with NaN: 

If I use the Month primitive on the following data: 

`pd.Series([datetime(2020, 3, 3), np.nan]) 
`
I get `[3.0, nan]`

If I pass `pd.Series([np.nan])` in, I get an error. 

I get analogous behavior for Year() and the like. I just wanted to confirm that this is indeed the correct expected behavior when writing tests.

Reply:
You should always expect the series dtype for `Datetime` columns to be `datetime64[ns]` based on the behavior of Woodwork, which enforces that dtype for the series.

If you don't otherwise specify the dtype a series of `pd.Series([np.nan])` will have a dtype of `float64` and you can't perform datetime operations on a `float64` column.

I haven't tested this out, but I think if you convert `pd.Series([np.nan])` to a datetime column first, the `Month` primitive will no longer error.

Reply:
Good call.  I confirmed this doesn't error
```python
s = pd.Series([np.nan]).astype('datetime64[ns]')

ft.primitives.Month()(s)
```